Minio/Helper/utils.cs

```diff
@@ -235,12 +235,15 @@ public static bool IsValidExpiry(int expiryInt)
         return expiryInt > 0 && expiryInt <= Constants.DefaultExpiryTime;
     }
 
-    internal static string getMD5SumStr(byte[] key)
+    internal static string getMD5SumStr(byte[] key, int exactBodySize = 0)
     {
+        var keyCopy = new byte[exactBodySize];
```

Comment:
Have you considered using an ArrayPool<byte> from `System.Buffers`?

```cs
        var keyCopy = ArrayPool<byte>.Shared.Rent(exactBodySize);

        try
        {
            var hashedBytes = MD5
                .Create()
                .ComputeHash(key);

            return Convert.ToBase64String(hashedBytes);
        }
        finally
        {
            ArrayPool<byte>.Shared.Return(keyCopy);
        }
```

Reply:
The problem with `ArrayPool` is that it does not give you the exact size you've requested, e.g. if you request an array of size `13`, the pool may give you probably a 16 bytes array or a bigger size depending on pool's availability. Hence, the `exactBodySize` variable is set to keep track of the correct size. If the rented array comes in the exact desired size, then the `exactBodySize` vaiable is set to `0`.
The bottom line is; returning an array by `ArrayPool` in exact/requested size is not guaranteed. It is a drawback in some cases, but it is one of the reasons `ArrayPool` has a smaller/better memory foot print. 

In our case, we don't mind carrying the content in a bigger size array, but only until the time when we calculate the authentication signature for the http request. So, at the latest point just before the http request's authentication signature is calculated, we had to create the correct sized array somehow; either in the memory as a byte[] array or on the disk whenever it is possible. 

The code you've commented on here is where we calculate the `md5sum` of the http content. So, this is where we need the exact sized byte array.